TanukiKit/Repositories.swift

```diff
@@ -5,60 +5,459 @@ import RequestKit
+     - parameter search: Return list of authorized projects matching the search criteria. Default is `""`
+     - parameter simple: Return only the ID, URL, name, and path of each project. Default is `""`, set to `true` to enable.
      - parameter completion: Callback for the outcome of the fetch.
      */
-    public func repositories(_ session: RequestKitURLSession = URLSession.shared, page: String = "1", perPage: String = "100", completion: @escaping (_ response: Response<[Repository]>) -> Void) -> URLSessionDataTaskProtocol? {
-        let router = RepositoryRouter.readAuthenticatedRepositories(configuration, page, perPage)
+
+    // TODO: projects convertion.
+    public func repositories(_ session: RequestKitURLSession = URLSession.shared, page: String = "1", perPage: String = "20", archived: String = "", visibility: String = "", orderBy: String = "created_at", sort: String = "desc", search: String = "", simple: String = "", completion: @escaping (_ response: Response<[Repository]>) -> Void) -> URLSessionDataTaskProtocol? {
+        let router = ProjectRouter.readAuthenticatedProjects(configuration: configuration, page: page, perPage: perPage, archived: archived, visibility: visibility, orderBy: orderBy, sort: sort, search: search, simple: simple)
         return router.loadJSON(session, expectedResultType: [[String: AnyObject]].self) { json, error in
             if let error = error {
                 completion(Response.failure(error))
             }
 
             if let json = json {
+                // TODO: Change to projects.
                 let repos = json.map { Repository($0) }
                 completion(Response.success(repos))
             }
         }
     }
+
+    /**
+     Fetches project for a specified ID.
+     - parameter id: The ID or NAMESPACE/PROJECT_NAME of the project.
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+
+    // TODO: project convertion.
+    public func repository(_ session: RequestKitURLSession = URLSession.shared, id: String, completion: @escaping (_ response: Response<Repository>) -> Void) -> URLSessionDataTaskProtocol? {
+        let router = ProjectRouter.readSingleProject(configuration: configuration, id: id)
+        return router.loadJSON(session, expectedResultType: [String: AnyObject].self) { json, error in
+            if let error = error {
+                completion(Response.failure(error))
+            }
+
+            if let json = json {
+                let project = Repository(json)
+                completion(Response.success(project))
+            }
+        }
+    }
+
+    /**
+     Fetches the Projects which the authenticated user can see.
+     - parameter page: Current page for project pagination. `1` by default.
+     - parameter perPage: Number of projects per page. `100` by default.
+     - parameter archived: Limit by archived status. Default is `""`, set to `true` to enable.
```

Comment:
Please use a `Bool` parameter instead of a String and default to the GitLab default.